lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -157,6 +157,11 @@ sub _setup_websocket_connection ($self, $websocket_url = undef) {
 
                 my $error = $tx->error;
                 my $error_message = "Unable to upgrade to ws connection via $websocket_url";
+
+                if ($tx->res->error && $tx->res->error->{message}) {
+                    $error_message .= sprintf(", %s in %s", $tx->res->error->{message}, $tx->req->url);
+                }
+
                 $error_message .= ", code $error->{code}" if ($error && $error->{code});
```

Comment:
You can just use `$error` for simplicity and there should always be a message: 
```suggestion
                if ($error) {
                    $error_message .= ", $error->{message}";
                    $error_message .= ", code $error->{code}" if $error->{code};
                }
```
I have also removed the URL because it is already part of the error message and I would just change that other part. However, if you disagree and want the original URL still being logged that's fine as well.